libs/common/src/types/database-connection.ts

```diff
@@ -0,0 +1,29 @@
+import { Db as DbConnection, MongoClient } from 'mongodb';
+
+/**
+ * Represents the status of a MongoDB client connection.
+ * This is a union type.
+ *
+ * This type can be either uninitialised or initialised.
+ */
+export type ClientConnectionStatus =
+  | {
+      /**
+       * Indicates that the MongoDB client connection is not initialised.
+       */
+      isInitialised: false;
+    }
+  | {
+      /**
+       * Indicates that the MongoDB client connection is initialised.
+       */
+      isInitialised: true;
+      /**
+       * The MongoDB client instance.
+       */
+      client: MongoClient;
+      /**
+       * The database connection instance.
+       */
+      connection: DbConnection;
+    };
```

Comment:
Should we include mongo in the file and type name instead of `database-connection` / `ClientConnectionStatus`? So it's clear it's for mongo

Also should we match the file name with the type name?